### PR TITLE
[MM-13837] Updated marked dependency

### PR DESCRIPTION
**Pull request by hahmadia**
#### Summary
Updates the `Marked` dependency

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/10774

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "localforage": "1.7.3",
     "localforage-observable": "2.0.0",
     "mark.js": "8.11.1",
-    "marked": "github:mattermost/marked#3c10e2109527cd14a76e1f8edd9d317d8b8988b0",
+    "marked": "github:mattermost/marked#3a9d0a5bd6d735df1a716ab50f73d58a1d865bdb",
```
Comment (saturninoabril) [inline package.json]: I don't see a test in marked's commit.  Please add there or here in `/link_only_renderer.test.jsx`, something like:
```javascript
{
    description: 'link: link with curly brackets',
    inputText: 'Let\'s try http://example/result?things={stuff}',
    outputText: 'Let&#39;s try <a class="theme markdown__link" href="http://example/result?things={stuff}" target="_blank">http://example/result?things={stuff}</a>',
},
```

Comment (hahmadia) [reply package.json]: Thanks for your suggestion. Test case has been added!

**Comment by hmhealey** [comment]
@hahmadia I'm not sure why the last build failed so I'm going to try re-running it. If it fails again, could you try merging the latest master into this branch? We've had some changes to the build system lately, so you might need something from the latest master to build properly

**Comment by hahmadia** [comment]
@hmhealey Seems like it worked out after your retry. Thanks for the help.

**Pull request merged**